cmd/kubeadm/app/preflight/checks.go

```diff
@@ -402,8 +403,12 @@ func (HostnameCheck) Name() string {
 }
 
 // Check validates if hostname match dns sub domain regex.
+// Check hostname length and format
 func (hc HostnameCheck) Check() (warnings, errorList []error) {
-	klog.V(1).Infoln("checking whether the given node name is reachable using net.LookupHost")
+	klog.V(1).Infoln("checking whether the given node name is valid and reachable using net.LookupHost")
+	for _, msg := range validation.IsQualifiedName(hc.nodeName) {
+		warnings = append(warnings, errors.Errorf("invalid format hostname \"%s\": %s. Change hostname or use --node-name flag to override it", hc.nodeName, msg))
```

Comment:
```suggestion
		warnings = append(warnings, errors.Errorf("invalid node name format %q: %s", hc.nodeName, msg))
```

requesting this since
- the kubeadm config uses node name under NodeRegistrationOptions (not hostName).
- the future of the --node-name flag is not clear so we should avoid mentioning it. it's GA, but we may remove it at some point.

Reply:
Done